controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -439,6 +440,26 @@ private long findSegmentSplitsMerges(List<Segment> referenceSegmentsList, List<S
                                                         segmentIndex.getData(), segmentTable.getData(), from, to)))));
     }
 
+    @Override
+    public CompletableFuture<Boolean> isStreamCutValid(Map<Long, Long> streamCut) {
+        return Futures.allOfWithResults(streamCut.keySet().stream().map(x -> getSegment(x).thenApply(segment ->
+                new SimpleEntry<>(segment.getKeyStart(), segment.getKeyEnd())))
+                .collect(Collectors.toList()))
+                .thenAccept(TableHelper::validateStreamCut)
+                .handle((r, e) -> {
+                    if (e != null) {
+                        if (Exceptions.unwrap(e) instanceof IllegalArgumentException) {
```

Comment:
Ideally we wouldn't be throwing illegal argument in this case. Can validate be made to return a result rather than throwing?

Reply:
We will address this comment as part of pull request where we make changes to validation logic based on new representation of stream cut.